### PR TITLE
Improvement/conversion flag

### LINKED ISSUE #100: Integrate ```-Wconversion``` compile flag
Add ```-Wconversion``` compile flag to the Makefile and remove all the generated warning messages.

### PR DESCRIPTION
**Pull request by b-carr**
This merge updates the kmyth submodule to the latest version, adds the ```-Wconversion``` flag for the compiler, and ensures there are no dicey type conversions. It will close #100.

### CODE REVIEW

**Review by bit-bender — COMMENTED** [review]
I have no comments about the type conversion fixes or kmyth version update.

I did notice three instances, however, where memory appears to be allocated and then immediately used without first checking that the memory allocation was successful.

File: src/util/cmd_interface.c [hunk 1]
```diff
@@ -118,34 +119,46 @@ static int msg_cmd(char *pipe, char *msg)
...
   int ret;
-  char *msg = (char *) calloc((10 + pipe_len + arg_len), sizeof(char));
 
-  sprintf(msg, "pelz %d %.*s %.*s", cmd, pipe_len, pipe, arg_len, arg);
+  // Due to the way we encode messages pipe_len and arg_len can't be too big.
+  if(pipe_len > INT_MAX || arg_len > INT_MAX)
+  {
+    return 1;
+  }
+  char *msg = (char *) calloc((10 + pipe_len + arg_len), sizeof(char));
```
[the remark this answers is not in the document]
Comment (b-carr) [reply src/util/cmd_interface.c:131]: Fixed in 5052ed3

File: src/util/cmd_interface.c [hunk 2]
```diff
@@ -118,34 +119,46 @@ static int msg_cmd(char *pipe, char *msg)
...
+int msg_list(char *pipe, size_t pipe_len, int cmd)
 {
   int ret;
+
+  // Due to the way we encode messages pipe_len can't be too big.
+  if(pipe_len > INT_MAX)
+  {
+    return 1;
+  }
   char *msg = (char *) calloc((10 + pipe_len), sizeof(char));
```
[the remark this answers is not in the document]
Comment (b-carr) [reply src/util/cmd_interface.c:163]: Fixed in 5052ed3

File: src/util/pipe_io.c [hunk 3]
```diff
@@ -104,8 +104,8 @@ int read_from_pipe(const char *pipe, char **msg)
 
   if (ret > 0)
   {
-    *msg = (char *) calloc(ret + 1, sizeof(char));
-    memcpy(*msg, buf, ret);
+    *msg = (char *) calloc((size_t)ret + 1, sizeof(char));
```
[the remark this answers is not in the document]
Comment (b-carr) [reply src/util/pipe_io.c:107]: Fixed in 5052ed3

**Pull request merged**